pandas/tests/arrays/categorical/test_repr.py

```diff
@@ -545,3 +545,11 @@ def test_categorical_str_repr(self):
         result = repr(Categorical([1, "2", 3, 4]))
         expected = "[1, '2', 3, 4]\nCategories (4, object): [1, 3, 4, '2']"
         assert result == expected
+
+    def test_categorical_with_pandas_series(self):
+        # GH 63045
+        s = Series(["apple", "banana", "cherry", "cherry"], dtype="string")
```

Comment:
```suggestion
    def test_categorical_with_pandas_series(self, string_dtype_no_object):
        # GH 63045
        s = Series(["apple", "banana", "cherry", "cherry"], dtype=string_dtype_no_object)
```

You could maybe use here this fixture that will test it for the different string dtype variations, to make sure we now do this consistently for all string like dtypes. 

The only thing you will have to update is the "string" in the expected result below (you can include `str(string_dtype_no_object)` in the expected value)